x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/enrollment/EnrollmentSingleNodeTests.java

```diff
@@ -0,0 +1,124 @@
+
+    @BeforeClass
+    public static void muteInFips() {
+        assumeFalse("Enrollment is not supported in FIPS 140-2 as we are using PKCS#12 keystores", inFipsJvm());
+    }
+
+    @Override
+    protected Settings nodeSettings() {
+        Settings.Builder builder = Settings.builder().put(super.nodeSettings());
+        addSSLSettingsForStore(
+            builder,
+            "xpack.security.http.",
+            "/org/elasticsearch/xpack/security/transport/ssl/certs/simple/httpCa.p12",
+            "password",
+            false
+        );
+        builder.put("xpack.security.http.ssl.enabled", true).put(ENROLLMENT_ENABLED.getKey(), "true");
+        // Need at least 2 threads because enrollment token creator internally uses a client
+        builder.put(EsExecutors.NODE_PROCESSORS_SETTING.getKey(), 2);
+        return builder.build();
+    }
+
+    @Override
+    protected boolean addMockHttpTransport() {
+        return false; // enable http
+    }
+
+    @Override
+    protected boolean transportSSLEnabled() {
+        return true;
+    }
+
+    public void testKibanaEnrollmentTokenCreation() throws Exception {
+        final SSLService sslService = getInstanceFromNode(SSLService.class);
+
+        final InternalEnrollmentTokenGenerator internalEnrollmentTokenGenerator = spy(
+            new InternalEnrollmentTokenGenerator(
+                newEnvironment(Settings.builder().put(ENROLLMENT_ENABLED.getKey(), "true").build()),
+                sslService,
+                node().client()
+            )
+        );
+        // Mock the getHttpsCaFingerprint method because the real method requires createClassLoader permission
+        Mockito.doReturn("fingerprint").when(internalEnrollmentTokenGenerator).getHttpsCaFingerprint();
+
+        final SetOnce<EnrollmentToken> enrollmentTokenSetOnce = new SetOnce<>();
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        // Create the kibana enrollment token and wait for the process to complete
+        internalEnrollmentTokenGenerator.createKibanaEnrollmentToken(enrollmentToken -> {
+            enrollmentTokenSetOnce.set(enrollmentToken);
+            latch.countDown();
+        }, List.of(TimeValue.timeValueMillis(500)).iterator());
+        latch.await(20, TimeUnit.SECONDS);
+
+        // The API key is created by the right user and should work
+        final Client apiKeyClient = client().filterWithHeader(
+            Map.of(
+                "Authorization",
+                "ApiKey " + Base64.getEncoder().encodeToString(enrollmentTokenSetOnce.get().getApiKey().getBytes(StandardCharsets.UTF_8))
+            )
+        );
+        final AuthenticateResponse authenticateResponse1 = apiKeyClient.execute(
+            AuthenticateAction.INSTANCE,
+            new AuthenticateRequest("_xpack_security")
+        ).actionGet();
+        assertThat(authenticateResponse1.authentication().getUser().principal(), equalTo("_xpack_security"));
```

Comment:
I think this is actually wrong. I think it would be better if node enrollment keys were owned by a "_node_enrollment" user rather than `_xpack_security`.

But it is what it is.

Reply:
I think so too, https://github.com/elastic/elasticsearch/pull/72186/files#r635377589 .